pkg/reconciler/domainmapping/table_test.go

```diff
@@ -0,0 +1,143 @@
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	servingclient "knative.dev/serving/pkg/client/injection/client/fake"
+	domainmappingreconciler "knative.dev/serving/pkg/client/injection/reconciler/serving/v1alpha1/domainmapping"
+	"knative.dev/serving/pkg/reconciler/domainmapping/resources"
+
+	. "knative.dev/pkg/reconciler/testing"
+	. "knative.dev/serving/pkg/reconciler/testing/v1"
+)
+
+func TestReconcile(t *testing.T) {
+	table := TableTest{{
+		Name: "bad workqueue key",
+		// Make sure Reconcile handles bad keys.
+		Key: "too/many/parts",
+	}, {
+		Name: "key not found",
+		// Make sure Reconcile handles good keys that don't exist.
+		Key: "foo/not-found",
+	}, {
+		Name: "first reconcile",
+		Key:  "default/first-reconcile.com",
+		Objects: []runtime.Object{
+			DomainMapping("default", "first-reconcile.com", WithRef("default", "target")),
+		},
+		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: DomainMapping("default", "first-reconcile.com",
+				WithRef("default", "target"),
+				WithURL("http", "first-reconcile.com"),
+				WithAddress("http", "first-reconcile.com"),
+				WithInitDomainMappingConditions,
+			),
+		}},
+		WantCreates: []runtime.Object{
+			resources.MakeIngress(DomainMapping("default", "first-reconcile.com", WithRef("default", "target")), "istio.ingress.networking.knative.dev"),
+		},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeNormal, "Created", "Created Ingress %q", "first-reconcile.com"),
+		},
+	}, {
+		Name: "reconcile changed ref",
+		Key:  "default/ingress-exists.org",
+		Objects: []runtime.Object{
+			DomainMapping("default", "ingress-exists.org", WithRef("default", "changed")),
+			resources.MakeIngress(DomainMapping("default", "ingress-exists.org", WithRef("default", "target")), "istio.ingress.networking.knative.dev"),
+		},
+		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: DomainMapping("default", "ingress-exists.org",
+				WithRef("default", "changed"),
+				WithURL("http", "ingress-exists.org"),
+				WithAddress("http", "ingress-exists.org"),
+				WithInitDomainMappingConditions,
+			),
+		}},
+		WantUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: resources.MakeIngress(DomainMapping("default", "ingress-exists.org", WithRef("default", "changed")), "istio.ingress.networking.knative.dev"),
+		}},
+	}}
+
+	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
+		r := &Reconciler{
+			netclient:     networkingclient.Get(ctx),
+			ingressLister: listers.GetIngressLister(),
+		}
+
+		return domainmappingreconciler.NewReconciler(ctx, logging.FromContext(ctx), servingclient.Get(ctx),
+			listers.GetDomainMappingLister(), controller.GetEventRecorder(ctx), r,
+		)
+	}))
+}
+
+type DomainMappingOption func(dm *v1alpha1.DomainMapping)
```

Comment:
Does this have to be public?

Reply:
Not really, was being consistent with [IngressOption](https://github.com/knative/serving/blob/138c6b739378a398604fb4d3b21cca607f429073/pkg/testing/functional.go#L336) but that's actually legit in another package. I'll lowercase it.